apps/andi/lib/andi/input_schemas/datasets/extract_steps/extract_step.ex

```diff
@@ -60,7 +60,7 @@ defmodule Andi.InputSchemas.Datasets.ExtractStep do
 
   def preload(struct), do: StructTools.preload(struct, [])
 
-  defp validate_type(%{changes: %{type: type}} = changeset) when type in ["http", "date"] do
+  defp validate_type(%{changes: %{type: type}} = changeset) when type in ["http", "date", "secret"] do
```

Comment:
Worth noting that Ecto (as of my recent merge) has an `Ecto.Enum` type that works for these kinds of type selections. Not expecting us to have to move to that, but worth noting